pr_agent/git_providers/github_provider.py

```diff
@@ -428,6 +428,130 @@ def publish_inline_comments(self, comments: list[dict], disable_fallback: bool =
             except Exception as e:
                 get_logger().error(f"Failed to publish inline code comments fallback, error: {e}")
                 raise e
+        
+    def get_review_comment_by_id(self, comment_id: int):
+        """
+        Retrieves a review comment by its ID.
+        
+        Args:
+            comment_id: Review comment ID
+            
+        Returns:
+            Review comment object or None (if not found)
+        """
+        try:
+            # Using PyGitHub library
+            # There's no direct way to get PR comment by ID, so we fetch all comments and filter
+            all_comments = list(self.pr.get_comments())
+            for comment in all_comments:
+                if comment.id == comment_id:
+                    return comment
+            return None
+        except Exception as e:
+            get_logger().warning(f"Failed to get review comment {comment_id}, error: {e}")
+            return None
+            
+    def get_review_id_by_comment_id(self, comment_id: int):
+        """
+        Finds the review ID that a comment belongs to based on its comment ID.
+        
+        Args:
+            comment_id: Review comment ID
+            
+        Returns:
+            Review ID or None (if not found)
+        """
+        try:
+            comment = self.get_review_comment_by_id(comment_id)
+            if comment:
+                return getattr(comment, 'pull_request_review_id', None)
+            return None
+        except Exception as e:
+            get_logger().warning(f"Failed to get review ID for comment {comment_id}, error: {e}")
+            return None
+    
+    def get_review_thread_comments(self, comment_id: int):
```

Comment:
Please make an effort to simplify the code complexity and runtime of this method. To collect a single thread discussion, a simpler solution is feasible.